File: service/src/main/java/net/huray/backend/minuting/config/FeignConfig.java
package net.huray.backend.minuting.config;

import feign.Retryer;
import feign.Retryer.Default;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.cloud.openfeign.EnableFeignClients;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@EnableFeignClients(basePackages = "net.huray.backend.minuting")
@Configuration
public class FeignConfig {

    @Value("${feign.retry.period}")
    private long period;

    @Value("${feign.retry.max-period}")
    private long maxPeriod;

    @Value("${feign.retry.max-attempts}")
    private int maxAttempts;

    @Bean
    public Retryer retryer() {
        return new Default(period, maxPeriod, maxAttempts);
    }

}

File: service/src/main/kotlin/net/huray/backend/minuting/config/FeignConfig.kt
package net.huray.backend.minuting.config

import feign.Retryer
import net.huray.backend.minuting.properties.FeignProperties
import org.springframework.cloud.openfeign.EnableFeignClients
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@EnableFeignClients(basePackages = ["net.huray.backend.minuting"])
@Configuration
class FeignConfig(
    private val feignProperties: FeignProperties
) {

    @Bean
    fun retryer(): Retryer = with(feignProperties) {
        Retryer.Default(period, maxPeriod, maxAttempts)
    }
    
}